### PR TITLE
Speedy: Cleanup and simplify code around userList, redirects, and function calls, and add plain G8 to categories

### PR DESCRIPTION
**Pull request by DannyS712**
Commits 1, 2, 4, 8 - tweak what criteria are shown where and how they are implemented
Commit 3 - wording tweak
Commits 5, 9-11 - reduce repeated function calls
Commit 6, 7 - change where the CSD menu is available

### CODE REVIEW

File: modules/twinklespeedy.js [hunk 1]
```diff
@@ -324,13 +330,13 @@ Twinkle.speedy.callback.modeChanged = function twinklespeedyCallbackModeChanged(
...
 
-	if (Twinkle.speedy.mode.isSysop(mode) && !Twinkle.speedy.mode.isMultiple(mode)) {
+	if (isSysop && !Twinkle.speedy.mode.isMultiple(mode)) {
 		work_area.append( { type: 'header', label: 'Custom rationale' } );
 		work_area.append( { type: radioOrCheckbox, name: 'csd', list: Twinkle.speedy.generateCsdList(Twinkle.speedy.customRationale, mode) } );
 	}
 
-	if (namespace % 2 === 1 && namespace !== 3) {
-		// show db-talk on talk pages, but not user talk pages
+	if (namespace % 2 === 1 && namespace !== 3 && namespace !== 447) {
```
[the remark this answers is not in the document]
Comment (DannyS712) [reply modules/twinklespeedy.js]: Done

File: modules/twinklespeedy.js [hunk 2]
```diff
@@ -22,12 +22,16 @@
 Twinkle.speedy = function twinklespeedy() {
 	// Disable on:
 	// * special pages
+	// * gadget pages
 	// * non-existent pages
-	if (mw.config.get('wgNamespaceNumber') < 0 || !mw.config.get('wgArticleId')) {
+	// * mediawiki pages (for non-sysops only)
+	var pageNamespace = mw.config.get('wgNamespaceNumber');
+	var isSysop = Morebits.userIsInGroup('sysop');
+	if (pageNamespace < 0 || !mw.config.get('wgArticleId') || pageNamespace === 2300 || (!isSysop && pageNamespace === 8) ) {
```
[the remark this answers is not in the document]
Comment (DannyS712) [reply modules/twinklespeedy.js]: Done

File: modules/twinklespeedy.js [hunk 3]
```diff
@@ -290,8 +295,9 @@ Twinkle.speedy.callback.modeChanged = function twinklespeedyCallbackModeChanged(
 
 	// first figure out what mode we're in
 	var mode = Twinkle.speedy.callback.getMode(form);
+	var isSysop = Twinkle.speedy.mode.isSysop(mode);
```
[the remark this answers is not in the document]
Comment (DannyS712) [reply modules/twinklespeedy.js]: Done

**Comment by DannyS712** [comment]
@Amorymeltzer I've actioned all of your requested changes (sorry for the confusing commit history, I was trying something out and misclicked)

File: modules/twinklespeedy.js [hunk 4]
```diff
@@ -346,15 +348,15 @@ Twinkle.speedy.callback.modeChanged = function twinklespeedyCallbackModeChanged(
...
 				break;
 
 			case 6:  // file
 			case 7:  // file talk
 				work_area.append( { type: 'header', label: 'Files' } );
 				work_area.append( { type: radioOrCheckbox, name: 'csd', list: Twinkle.speedy.generateCsdList(Twinkle.speedy.fileList, mode) } );
-				if (!Twinkle.speedy.mode.isSysop(mode)) {
-					work_area.append( { type: 'div', label: 'Tagging for CSD F4 (no license), F5 (orphaned fair use), F6 (no fair use rationale), and F11 (no permission) can be done using Twinkle\'s "DI" tab.' } );
+				if (!isSysopMode) {
+					work_area.append( { type: 'div', label: 'Tagging for CSD F4 (no license), F5 (orphaned fair use), F6 (no fair use rationale), and F11 (no permission) is done using Twinkle\'s "DI" tab.' } );
```
Comment (Amorymeltzer) [inline modules/twinklespeedy.js]:
```suggestion
					work_area.append( { type: 'div', label: 'Tagging for CSD F4 (no license), F5 (orphaned fair use), F6 (no fair use rationale), and F11 (no permission) can be done using Twinkle\'s "DI" tab.' } );
```
This is commit 27bf3fa, but I think the previous wording was clearer, and the commit message suggests this was made under a misunderstanding.

**Pull request merged**